base/replutil.jl

```diff
@@ -108,9 +108,12 @@ function showerror(io::IO, ex::DomainError, bt; backtrace=true)
             if code[1] in (:log, :log2, :log10, :sqrt) # TODO add :besselj, :besseli, :bessely, :besselk
                 println(io,"\n$(code[1]) will only return a complex result if called with a complex argument.")
                 print(io, "try $(code[1]) (complex(x))")
-            elseif (code[1] == :^ && code[2] == symbol("intfuncs.jl")) || code[1] == :power_by_squaring
+            elseif (code[1] == :^ && code[2] == symbol("intfuncs.jl")) || code[1] == :power_by_squaring #3024
                 println(io, "\nCannot raise an integer x to a negative power -n. Make x a float by adding")
-                print(io, "a zero decimal (e.g. 2.0^-n instead of 2^-n), or write 1/x^n, float(x)^-n, or (x//1)^-n")
+                print(io, "a zero decimal (e.g. 2.0^-n instead of 2^-n), or write 1/x^n, float(x)^-n, or (x//1)^-n .")
+            elseif code[1] == :^ && (code[2] == symbol("promotion.jl") || code[2] == symbol("math.jl"))
+                println(io, "\nCannot raise an integer or float yielding a complex number. ")
+                print(io, "If x is the input number that should be raised to y, use (x+0im)^y or convert(Complex, x)^y .")
```

Comment:
Stylistically, it is weird that we are using multiple `print` calls here.  Would be cleaner to replace this (and similar constructions above) with:
```jl
print(io, "\n ............\n",
        ".......")
```